cron-jobs/check_archlinux/parse_pkgbuilds.sh

```diff
@@ -6,23 +6,23 @@
 exit() { return; }
 
 splitpkg_overrides=('depends' 'optdepends' 'provides' 'conflicts')
-variables=('pkgname' 'pkgbase' 'epoch' 'pkgver' 'pkgrel' 'makedepends' 'arch' ${splitpkg_overrides[@]})
+variables=('pkgname' 'pkgbase' 'epoch' 'pkgver' 'pkgrel' 'makedepends' 'arch' "${splitpkg_overrides[@]}")
 readonly -a variables splitpkg_overrides
 
 backup_package_variables() {
-	for var in ${splitpkg_overrides[@]}; do
+	for var in "${splitpkg_overrides[@]}"; do
 		indirect="${var}_backup"
-		eval "${indirect}=(\${$var[@]})"
+		eval "${indirect}=(\"\${$var[@]}\")"
 	done
 }
 
 restore_package_variables() {
-	for var in ${splitpkg_overrides[@]}; do
+	for var in "${splitpkg_overrides[@]}"; do
 		indirect="${var}_backup"
 		if [ -n "${!indirect}" ]; then
-			eval "${var}=(\${$indirect[@]})"
+			eval "${var}=(\"\${$indirect[@]}\")"
```

Comment:
I would still like to fix this by avoiding the issue entirely and using `declare` as I mentioned on the mailing list.

Reply:
1. From the ML:

> Maybe I should fix the backups as well, but that is a slightly more complicated case there [in makepkg].

Is it?  It looks like the exact same case there.

2. What about this?

```
backup_package_variables() {
	restore_package_variables=$(declare -p "${splitpkg_overrides[@]}" 2>/dev/null || true)
}

restore_package_variables() {
	unset "${splitpkg_overrides[@]}"
	eval "$restore_package_variables"
}
```